src/main/java/main/ResponseFactory.java

```diff
@@ -12,23 +13,25 @@
 public class ResponseFactory {
     private final HashMap<String, DefaultResponse> responses;
```

Comment:
Make sure you use interface types where necessary. I.e. `private final Map<Foo, Bar> r; r = new HashMap<>`